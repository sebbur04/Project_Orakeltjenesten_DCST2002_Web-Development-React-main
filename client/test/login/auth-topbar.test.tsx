// Import neccessary modules for testing
import * as React from "react";
import { shallow } from "enzyme";
import TopBar from "../../src/login/topbar";

jest.mock("../../src/service/user-service", () => ({
  logout: jest.fn().mockResolvedValue(undefined),
}));

describe("TopBar component tests", () => {
  test("TopBar renders correctly", (done) => {
    const wrapper = shallow(<TopBar />);

    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });

  test("Orakeltjenesten Button takes you to home", (done) => {
    const wrapper = shallow(<TopBar />);

    wrapper.find("a").at(0).simulate("click");

    setTimeout(() => {
      expect(location.hash).toEqual("#/");
      done();
    });
  });
});
