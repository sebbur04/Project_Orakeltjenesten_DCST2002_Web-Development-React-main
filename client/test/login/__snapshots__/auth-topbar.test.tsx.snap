// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar component tests TopBar renders correctly 1`] = `
<div
  className="top-container"
>
  <div
    className="title"
  >
    <a
      className="orakel-title"
      href="/#/"
    >
      Orakeltjenesten
    </a>
  </div>
  <SearchBar />
  <div
    className="top-right-container"
  >
    <DarkModeButton />
    <div>
      <ButtonLight
        onClick={[Function]}
      >
        Profil
      </ButtonLight>
    </div>
    <ButtonDanger
      onClick={[Function]}
    >
      Logg ut
    </ButtonDanger>
  </div>
</div>
`;
